src/core/linter-rules/check-period-in-p.js

```diff
@@ -23,26 +23,12 @@ const lang = defaultLang in meta ? defaultLang : "en";
  * @param  {Document} doc The document to be checked.
  */
 function lintingFunction(conf, doc) {
-  console.log("dsfdf")
   
-  const allElements = Array.from(doc.querySelectorAll("p"));
-  console.log(allElements);
   const offendingElements = Array.from(doc.querySelectorAll("p"))
-    // this gives us all paragraphs that don't end with a period
-    .filter(elem => !elem.textContent.trim().endsWith("."))
-    .filter(elem => !elem.textContent.trim().endsWith(":"))
-    .filter(elem => !elem.textContent.trim().endsWith("?"))
-    .filter(elem => !elem.textContent.trim().endsWith("!"))
-    .filter(elem => (elem.textContent.trim() !== ""))
-    .filter(elem => !elem.textContent.trim().includes("tests:"))
-    
+    .filter((elem) => !(/[.!?:]$|^ *$/.test(elem.textContent.trim()))
```

Comment:
As regexes are super annoying to read, always put a comment above describing what it does.

Reply:
So, in theory, we shouldn't need .trim() there either. 
```
.filter(({textContent}) => !(/[.!?:]$|^ *$/.test())
```

Reply:
fixed mistake above

Reply:
ok, now we also want multiline... so we need "m":

.filter(({textContent}) => !(/[.!?:]$|^\s*$/m.test(textContent))